test/spec/videoCache_spec.js

```diff
@@ -297,6 +298,34 @@ describe('The video cache', function () {
       JSON.parse(request.requestBody).should.deep.equal(payload);
     });
 
+    it('should wait the duration of the batchTimeout and pass the correct batchSize if batched requests are enabled in the config', () => {
+      let callback = null;
+      let mockTimeout = sinon.stub().callsFake((cb) => { callback = cb });
+
+      config.setConfig({
+        cache: {
+          url: 'https://prebid.adnxs.com/pbc/v1/cache',
+          batchSize: 3,
+          batchTimeout: 20
+        }
+      });
+
+      let stubCache = sinon.stub();
+      const batchAndStore = batchingCache(mockTimeout, stubCache);
+      for (let i = 0; i < 3; i++) {
+        batchAndStore({}, {}, function() {});
+      }
+
+      sinon.assert.calledOnce(mockTimeout);
+      sinon.assert.calledWith(mockTimeout, sinon.match.any, 20);
+
+      const expectedBatch = [{ afterBidAdded: function () {}, auctionInstance: { }, bidResponse: { } }, { afterBidAdded: function () {}, auctionInstance: { }, bidResponse: { } }, { afterBidAdded: function () {}, auctionInstance: { }, bidResponse: { } }];
```

Comment:
different functions have different identities; `function(){} === function(){}` is always false. Same goes for `{} === {}`. You need to pass in the same objects if you want to test them for equality.

```javascript
const mockAfterBidAdded = function() {}
// ...
batchAndStore(mockAfterBidAdded, {}, {})
// ...

expectedBatch = [{afterBidAdded: mockAfterBidAdded, ...}, ...]
```

I think `assertCalledWith` does deep equality checks, so `{}` matches `{}` even if they are different objects - it just doesn't know how to do deep equals on function objects. If that's not the case you will need to do the same with `bidResponse` and `auctionInstance`.

Reply:
ah, you're explanation makes sense.  just tried your fix and it worked 👍 thanks!